### DISCUSSION TITLE
Enabling accelerated generation Mac

### DISCUSSION THREAD

**Discussion by edmundronald** [oobabooga/text-generation-webui#4831] [Q&A] [answered]
Hi,

What flags should I set, and especially how do I se them to speed up generation on my Mac M3 Max laptop with 128GB of RAM? 
I am using llama.cpp as a backend and the model is openorca-zephyr-7b.Q8_0.gguf.
I am not at the moment seeing the GPU stressed at all, in the mac's Activity Monitor tool.

Edmund

**Comment by oobabooga** [comment, marked as answer]
Make sure to set n_gpu_layers to more than 0 before loading the model.

**Reply by edmundronald** [reply]
How do I set this variable? 
I'm not having much luck with this command line

% ./start_macos.sh --n_gpu_layers 1

Edmund

**Reply by TheLounger** [reply]
It's `--n-gpu-layers` as a command-line argument (see [here](https://github.com/oobabooga/text-generation-webui#llamacpp)). Alternatively you can set it in `config-user.yaml` as `n_gpu_layers`, or just set it in the UI under the Model tab.

**Reply by edmundronald** [reply]
TheLounger

I made it work with both the slider and the command line arg. 
Thank you very much for your patient reply to someone who confuses dashes and underscores :). 

Edmund
